src/r0/directory/get_public_rooms_filtered.rs

```diff
@@ -55,3 +65,174 @@ pub struct Filter {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub generic_search_term: Option<String>,
 }
+
+/// Information about which networks/protocols from application services on the
+/// homeserver from which to request rooms.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum RoomNetwork {
+    /// Return rooms from the Matrix network.
+    Matrix,
+    /// Return rooms from all the networks/protocols the homeserver knows about.
+    All,
+    /// Return rooms from a specific third party network/protocol.
+    ThirdParty(String),
+}
+
+impl Default for RoomNetwork {
+    fn default() -> Self {
+        RoomNetwork::Matrix
+    }
+}
+
+impl Serialize for RoomNetwork {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state;
+        match *self {
+            Self::Matrix => {
+                state = serializer.serialize_struct("RoomNetwork", 0)?;
+            }
+            Self::All => {
+                state = serializer.serialize_struct("RoomNetwork", 1)?;
+                state.serialize_field("include_all_networks", &true)?;
+            }
+            Self::ThirdParty(ref network) => {
```

Comment:
... you can remove `ref` here:

```suggestion
            Self::ThirdParty(network) => {
```

Reply:
I think I tried that and it didn't work. I can check again.

Removing the dereference of self also requires you to add & references to all the match patterns.

Reply:
It really shouldn't require that, unless you're using a really old version of rust locally.